From 855f25a0c64f096fda99d14a5fdfa980fdeea8c1 Mon Sep 17 00:00:00 2001
From: Algea Cao <algea.cao@rock-chips.com>
Date: Wed, 28 Aug 2019 17:46:26 +0800
Subject: [PATCH] drm: bridge: synopsys: dw-hdmi: Fix kernel logo flash

Filling in the missing parts from:
'commit d7f22ab4c135 ("drm: bridge: dw-hdmi: fix wrong color if get
edid error at bootup")'

Change-Id: I92b690dcc332e9f9e50ef7d457734952a9f1ef2a
Signed-off-by: Algea Cao <algea.cao@rock-chips.com>
---
 drivers/gpu/drm/bridge/synopsys/dw-hdmi.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c b/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c
index 11b79d9e6daa..d3cdb52eeced 100644
--- a/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c
+++ b/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c
@@ -2883,6 +2883,19 @@ static void dw_hdmi_attach_properties(struct dw_hdmi *hdmi)
 			dev_err(hdmi->dev, "unexpected mapping: 0x%x\n",
 				video_mapping);
 		}
+
+		hdmi->hdmi_data.enc_in_bus_format = color;
+		hdmi->hdmi_data.enc_out_bus_format = color;
+		/*
+		 * input format will be set as yuv444 when output
+		 * format is yuv420
+		 */
+		if (color == MEDIA_BUS_FMT_UYVY10_1X20)
+			hdmi->hdmi_data.enc_in_bus_format =
+				MEDIA_BUS_FMT_YUV10_1X30;
+		else if (color == MEDIA_BUS_FMT_UYVY8_1X16)
+			hdmi->hdmi_data.enc_in_bus_format =
+				MEDIA_BUS_FMT_YUV8_1X24;
 	}
 
 	if (ops && ops->attach_properties)
-- 
2.35.3

